src/InsertionsClient/Api/Props/PropsVariableDeducer.cs

```diff
@@ -0,0 +1,210 @@
+                FindPackageByIdResource resource = await rep.GetResourceAsync<FindPackageByIdResource>().ConfigureAwait(false);
+
+                Trace.WriteLine($"Downloading package {packageUpdate.PackageId}-{packageUpdate.NewVersion} into memory.");
+
+                bool downloadResult = await resource.CopyNupkgToStreamAsync(
+                    packageUpdate.PackageId,
+                    packageVersion,
+                    packageStream,
+                    cache,
+                    NullLogger.Instance,
+                    cancellationToken ?? ownedCancellationToken!.Token)
+                    .ConfigureAwait(false);
+
+                if (!downloadResult)
+                {
+                    Trace.WriteLine($"There is an issue downloading nuget package: {packageUpdate.PackageId}");
+                    return Enumerable.Empty<string>();
+                }
+
+                Trace.WriteLine($"Downloading complete for package {packageUpdate.PackageId}.");
+                return GetNugetFileList(packageStream, link);
+            }
+            catch (Exception e)
+            {
+                Trace.WriteLine($"There is an issue downloading nuget package.");
+                Trace.WriteLine(e.ToString());
+                return Enumerable.Empty<string>();
+            }
+            finally
+            {
+                ownedCancellationToken?.Dispose();
+            }
+        }
+
+        private IEnumerable<PropsFileVariableReference> MatchFileNames(SwrFile[] swrFiles, string filename)
+        {
+            foreach (SwrFile swrFile in swrFiles)
+            {
+                foreach (PayloadPath payloadPath in swrFile.PayloadPaths)
+                {
+                    Match match = payloadPath.Pattern.Match(filename);
+                    if (!match.Success)
+                    {
+                        continue;
+                    }
+
+                    yield return new PropsFileVariableReference(payloadPath.VariableName, match.Groups[1].Value, swrFile.Path);
+                }
+            }
+        }
+    
+        /// <summary>
+        /// Retrieves the paths to all the files in the nuget package as if they were extracted to the location
+        /// specified in <see cref="directory"/> parameter.
+        /// </summary>
+        /// <param name="packageDataStream">Stream containing the nuget package binary.</param>
+        /// <param name="directory">Directory for the extracted files in the package.</param>
+        /// <returns></returns>
+        private List<string> GetNugetFileList(Stream packageDataStream, string directory)
+        {
+            using PackageArchiveReader packageReader = new PackageArchiveReader(packageDataStream);
+
+            return packageReader.GetFiles().Select(file =>
+            {
+                // Construct path to the file
+                // Link always has backslash. Make sure it is consistent.
+                return directory + "\\" + file.Replace('/', '\\');
```

Comment:
```suggestion
                return Path.Combine(directory, file.Replace('/', '\\'));
```

Reply:
Payload paths in .swr files always have `\`. If we don't do the same, string matching will fail.